### PR TITLE
Add scheduled job to check for new envelopes

### PR DESCRIPTION
**Pull request by rkondratowicz**
disabled for now

https://tools.hmcts.net/jira/browse/BPS-985

### CODE REVIEW

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTaskTest.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/config/TimeConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5250737&bid=17206893)

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private static final Logger log = LoggerFactory.getLogger(NoNewEnvelopesTask.class);
+
+    private final NewEnvelopesChecker checker;
+
+    public NoNewEnvelopesTask(NewEnvelopesChecker checker) {
+        this.checker = checker;
+    }
+
+    @Scheduled(cron = "0 0 * * * *", zone = EUROPE_LONDON)
+    @SchedulerLock(name = "no-new-envelopes", lockAtLeastFor = "30s")
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java]: Should we extract name to constant?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public void run() {
+        log.info("Starting no-new-envelopes job");
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java]: ..and use here and in finishing log?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java]: it is only done in blob-router :/

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        checker.checkIfEnvelopesAreMissing();
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java]: should we also log finishing of job ?

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTaskTest.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/config/TimeConfiguration.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5250737&bid=17206893)

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java (hunk 3 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/monitoring/NoNewEnvelopesTask.java]: fixed

**Pull request merged**